### PR TITLE
feature: add 'Domain' property to RegisterModel step

### PR DESCRIPTION
**Pull request by rahven14**
*Issue #, if available:*

*Description of changes:*

add 'Domain' property to RegisterModel step

*Testing done:*

unit
Command used: export IGNORE_COVERAGE=- ; tox -e py38 -- -s -vv tests/unit/test_session.py::test_create_model_package_from_containers_all_args ; unset IGNORE_COVERAGE

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -4309,6 +4318,7 @@ def get_create_model_package_request(
     drift_check_baselines=None,
     customer_metadata_properties=None,
     validation_specification=None,
+    domain=None
```
Comment (MadhubalasriB) [inline src/sagemaker/session.py:4321]: Missing domain description here

**Review by MadhubalasriB — APPROVED** [review]

File: src/sagemaker/estimator.py [hunk 2]
```diff
@@ -1311,6 +1312,8 @@ def register(
             drift_check_baselines (DriftCheckBaselines): DriftCheckBaselines object (default: None).
             customer_metadata_properties (dict[str, str]): A dictionary of key-value paired
                 metadata properties (default: None).
+            domain (str): Domain values can be "COMPUTER_VISION", "NATURAL_LANGUAGE_PROCESSING",
```
[the remark this answers is not in the document]
Comment (rahven14) [reply src/sagemaker/estimator.py:1315]: Yes, this is true. It has been verified from the CLI as well.

**Review by staubhp — APPROVED** [review]

File: src/sagemaker/session.py (hunk 1 above)
Comment (rahven14) [reply src/sagemaker/session.py:4321]: added the missing domain description

**Comment by qidewenwhen** [comment]
Out of curiosity, is there any reason we don't add the `Domain` to the register method of models like `HuggingFaceModel`, `SKLearnModel` and `PipelineModel`?

**Review by shreyapandit — APPROVED** [review]

**Review by qidewenwhen — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]

**Pull request merged**